### PR TITLE
    Fix negation of 0 length Decimal

### PR DESCRIPTION
**Pull request by mortenbekditlevsen**
A Decimal value with _length 0 and _isNegative set to 1 is interpreted as a NaN. The 'negate()' function however, flipped the _isNegative flag without regard for the _length 0 case. This meant that -0 would become NaN. The fix checks for the _length 0 special case. In NSDecimalSubtract() the same check was performed. Since this now happens in negate(), it is removed from the NSDecimalSubtract() function.

Please see the discussion for a similar pull request in swift-corelibs-foundation: https://github.com/apple/swift-corelibs-foundation/pull/1117

### CODE REVIEW

File: test/stdlib/TestDecimal.swift [hunk 1]
```diff
@@ -273,6 +273,9 @@ class TestDecimal : TestDecimalSuper {
         expectEqual(.minus, d.sign)
         d.negate()
         expectEqual(.plus, d.sign)
+        var e = Decimal(0)
+        e.negate()
+        XCTAssertEqual(e, 0)
```
Comment (phausler) [inline test/stdlib/TestDecimal.swift]: this should be `expectEqual` not `XCTAssertEqual` :/

Comment (mortenbekditlevsen) [inline test/stdlib/TestDecimal.swift]: Many apologies, I am having trouble building the swift repo.

**Review by phausler — APPROVED** [review]
Looks good to me, thanks again for fixing this

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by parkera** [comment]
@swift-ci smoke test and merge

**Comment by CodaFi** [comment]
Looks like Jenkins crashed mid-merge.  Doing it manually.

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test and merge

**Comment by CodaFi** [comment]
@shahmishal Any idea why the merge URL for this is 405'ing?

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
⛵️

**Comment by mortenbekditlevsen** [comment]
Huzzah! Thanks! :-)

**Pull request merged**